gui/backend/upsert_bot_config.go

```diff
@@ -178,6 +186,23 @@ func (s *APIServer) validateConfigs(req upsertBotConfigRequest) *upsertBotConfig
 	return nil
 }
 
+func isBotNameValid(botName string) bool {
+	if botNameRegex == nil {
+		panic(fmt.Errorf("botname regex undefined at runtime"))
+	}
+
+	return botNameRegex.MatchString(botName)
+}
+
+// InitBotNameRegex initializes the regex for bot names.
+func InitBotNameRegex() (e error) {
```

Comment:
for higher quality code we should prevent double initialization, i.e. this should throw an error if botNameRegex is non-nil at the beginning of this function.